app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,27 @@
+class CommentsController < ApplicationController
+  before_action :authenticate_user!
+
+  def new
+    @comment = Comment.new
+  end
+
+  def create
+    @comment = @commentable.comments.new(comment_params)
+    @comment.user = current_user
+    if @comment.save
+      if request.xhr?
+        redirect_to @commentable
+      else
+        redirect_to @commentable
+      end
+    else
+      redirect_to action: 'show', controller:'reviews' , id:params[:review][:id].to_i
```

Comment:
Redirect on a failed save - is that correct? Also, is there a path helper you could use here?